app/controllers/users_controller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# users_controller.rb
+class UsersController < ApplicationController
+  before_action :logged_in, only: :new
+  before_action :find_user, only: :show
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    flash.now[:alert] = 'There was a problem signing up.'
```

Comment:
You should set the error message only if the validation fails.

Reply:
Done !